mesonbuild/mparser.py

```diff
@@ -201,7 +201,7 @@ def lex(self, filename: str) -> T.Generator[Token, None, None]:
                         lines = value.split('\n')
                         if len(lines) > 1:
                             lineno += len(lines) - 1
-                            line_start = mo.end() - len(lines[-1])
+                            line_start = mo.end() - len(lines[-1]) - 3
```

Comment:
Where does this 3 come from? You should add explanation to the commit message, or as a comment.

Reply:
I updated the commit message.